### ISSUE TITLE
Better Content Warnings

### ISSUE THREAD

**Issue by neonpixii** [pixelfed/pixelfed#408]
As it stands the content warning system is not particularly effective; there's no field for entering in what the sensitive content is. You just get a generic message. 

Being able to change the message from "Content Warning: This may contain potentially sensitive content" to something more descriptive for each individual post would be a huge benefit for usablitiy.

Additionally, displaying the custom content warning text on the greyed-out thumbnail in the profile view too would help people find a content-warned image in a profile that has had heavy use better (though maybe that would be best put into a separate issue)

**Comment by trwnh** [comment]
Three aspects to this, both currently unfinished:

- [ ] Apply a "subject" (`summary` in ActivityPub)
- [ ] Hide/blur out thumbnails for sensitive posts
- [ ] Possibly show part of the "subject" on the obfuscated thumbnail? (idk if this can easily look good though)

**Comment by lapineige** [comment]
> Hide/blur out thumbnails for sensitive posts

I think a blur would be much better that the actual "No preview available" and "CW / NSFW / Hidden Media (click to show)", which is unattractive and a bit boring (I mean, its look and feel).

As Pixelfed is focused on image sharing, I think that fulling hiding the content (as it's done right now, or with a black background) is not the most interesting solution.
A blurred image, such as a blurhash, would be a lot more interesting : it would be way more attractive while still hiding the content.

And maybe the CW text could be displayed on top of that blurred image ? Then the viewer is fully and easily aware of the potential trigger, as well as the subject of the image.

---

PS: this is slightly off-topic (?), be could the user have an option to ignore all CW in its timeline or on a specific profile page/for a specific user ?
For instance: let's say one person without arachnophobia follow a photographer who publish only/mainly spiders pictures. Maybe that person would like to see all the pictures in the profile/in his or her TL, without having to disable all CW.
(obviously this would be opt-in)
With such an option (button ?) it would allow that person (only) to ignore the CW for that specific user, without disabling them globally. Hence it would provide him/her more comfort, without increasing her/his exposure to potential triggers from other people (for instance in the global TL).

**Comment by lapineige** [comment]
Another proposal: on profile page, add an option to display all media, including the hidden ones.
This way, people could *willingly* chose to ignore CW to display the profile pictures without the unaesthetic¹ hidden media.

¹ except maybe blurhash ones, at least it's not a dark a gray or black square or similar.
